bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java

```diff
@@ -35,94 +37,103 @@
  *
  * @author Osman Basha - Initial contribution
  */
-public class SMAEnergyMeterHandler extends BaseThingHandler {
+public class SMAEnergyMeterHandler extends BaseThingHandler implements PayloadHandler {
 
     private final Logger logger = LoggerFactory.getLogger(SMAEnergyMeterHandler.class);
-    private EnergyMeter energyMeter;
-    private ScheduledFuture<?> pollingJob;
+    private final PacketListenerRegistry listenerRegistry;
+    private @Nullable PacketListener listener;
+    private @Nullable String serialNumber;
 
-    public SMAEnergyMeterHandler(Thing thing) {
+    public SMAEnergyMeterHandler(Thing thing, PacketListenerRegistry listenerRegistry) {
         super(thing);
+        this.listenerRegistry = listenerRegistry;
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
         if (command == RefreshType.REFRESH) {
             logger.debug("Refreshing {}", channelUID);
-            updateData();
+            PacketListener listener = this.listener;
+            if (listener != null) {
+                listener.request();
+            }
         } else {
             logger.warn("This binding is a read-only binding and cannot handle commands");
         }
     }
 
     @Override
     public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Put this line rather at line 84, this will avoid a doubled thing state change in case one parameter is missing, UNKNOWN and finally OFFLINE few ms after.

Reply:
done